dhall-json/dhall-json.cabal

```diff
@@ -35,6 +35,11 @@ Source-Repository head
     Type: git
     Location: https://github.com/dhall-lang/dhall-haskell/tree/master/dhall-json
 
+Flag gpl
+  Description: Use GPL-licensed components like HsYAML, and enable yaml-to-dhall binary
+  Default: False
+  Manual: True
```

Comment:
Why a manual flag? Why not like Cabal choose whatever if it's unspecified? This opens the door to more potential build plans, if you don't care.

Reply:
That didn't seem like an *explicit* opt-in.

How can you not care about something that will cause you to have to relicense your project GPL? If you're using dhall-json as a library, that's something you have to consider. And if you're not, it's easy to do `cabal v2-install -fgpl dhall-json`.

I personally wouldn't expect a Dhall (which is otherwise BSD3) utility package to semi-randomly have this effect. Curious what others think.

Reply:
I agree with @patrickmn that this flag should definitely be manual (since we don't want `cabal`'s solver to govern the decision of what license to use) and off by default (since I imagine most users will prefer a BSD license but they can enable the GPL version if they better YAML support and are willing to convince management/customers that it's safe)

Reply:
But I say you *do* want Cabal to choose unless you actually care. But it seems like we disagree there, and that's ok - we've had the conversation which is more important

Reply:
FWIW there shouldn't be a case where aeson-yaml won't build; it only depends on aeson, bytestring, text, unordered-containers and vector.